### DISCUSSION TITLE
SSG: avoid duplicate request for multiple documents / enrich context ?

### DISCUSSION THREAD

**Discussion by svassaux** [vercel/next.js#27846] [Help]
Hello :)

I am using SSG for my website.
I have a help center and hundreds of articles.

On those articles I have a table of content for the full help center. 

In getStaticProps I make a paginated query to retrieve fragments of all those articles and build my table of content.
It is repeated for every article dispite the data is the same.

How can I avoid that? Can I enrich the context argument of the getStaticProp somehow? Any other way?

Hanks a lot

**Comment by thomasverleye** [comment]
Hi,

I'm afraid this is not possible if you want it to be working SSG.
But if you are okay with switching to SSR it could be possible to gather this data once if you use `getInitialProps` in your `_app.js`.

More info on that can be read on this [documentation page](https://nextjs.org/docs/advanced-features/custom-app).

I haven't implemented this myself, but it should work.

Good luck!
